Source/AwsCommonRuntimeKit/sdkutils/CRTAWSEndpointsResolvedEndpoint.swift

```diff
@@ -0,0 +1,93 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import Foundation
+import AwsCSdkUtils
+
+/// Resolved endpoint
+public class CRTAWSEndpointResolvedEndpoint {
+    let rawValue: OpaquePointer
+    
+    
+    /// Initialize a new resolved endpoint
+    /// - Parameter rawValue: The raw value of the resolved endpoint
+    public init(rawValue: OpaquePointer) {
+        self.rawValue = rawValue
+        
+        aws_endpoints_resolved_endpoint_acquire(rawValue)
+    }
+    
+    
+    /// Get the type of the resolved endpoint
+    /// - Returns: The type of the resolved endpoint
+    public func getType() -> CRTAWSEndpointsResolvedEndpointType {
+        let type = aws_endpoints_resolved_endpoint_get_type(rawValue)
+        return CRTAWSEndpointsResolvedEndpointType(rawValue: type)
+    }
+    
+    
+    /// Get the URL of the resolved endpoint
+    /// - Returns: The URL of the resolved endpoint
+    public func getURL() throws -> String? {
```

Comment:
Why return an optional String from a throwing function?  Is there a scenario where `nil` would be returned, and it is not an error?

Reply:
Yes, for whatever reason CRT return a non-zero response in Line33, here we want to bubble up the error thrown by the CRT than just return a nil.

Secondly, pointee.toString() returns a nullable string. Sending a default empty string is not correct and we will endup in some strange case in the consumer side.

Reply:
Agreed, returning an empty string in place of the nil doesn't solve anything, even if it allows the return type to be changed from `String?` to `String`.

What I'm asking is if we should be throwing an error instead of returning nil, i.e.
```
guard let urlString = urlOut.pointee.toString() else {
    throw CRTError.awsError(/* insert appropriate error here */)
}
return urlString
```

Because if returning `nil` without throwing an error is abnormal/unexpected, we should not permit it.  And if it isn't unexpected, then at a minimum our documentation comment for this function should say that it may be nil & why.

Reply:
It should be okay throwing error, it won't be `awsError` but rather a custom error. It would be good experience too.

Reply:
I was thinking more and trying to see options and convinced that returning `nil` is better here and let the caller handle the issue. Reason being, this is a binding that should just work as wrapper around the C code. Further handling of error/nil must be done in the next layer i.e. AWSClientRuntime.

This also aligns with the existing bindings such as https://github.com/awslabs/aws-crt-swift/blob/2565815e8963455d8298afa4d5a8a44544e2b03c/Source/AwsCommonRuntimeKit/auth/credentials/CRTCredentials.swift#L32

TLDR, keeping it it is.

Reply:
OK.  We can keep this for now if this follows the convention in the project.  But one of the functions of these bindings should be to enforce guarantees in Swift that can't be enforced in C, so that the output from CRT functions excludes undefined states like nil result and no error.

Put another way, if I wanted to test everything for `nil` before using it, I'd be a Java developer.  ;)